ui/pages/confirmations/components/confirm/header/header.tsx

```diff
@@ -33,7 +33,16 @@ const Header = () => {
 
   const { currentConfirmation } = useConfirmContext<Confirmation>();
 
-  if (currentConfirmation?.type === TransactionType.tokenMethodTransfer) {
+  const CONFIRMATIONS_WITH_NEW_HEADER = [
+    TransactionType.tokenMethodTransfer,
+    TransactionType.tokenMethodTransferFrom,
+    TransactionType.tokenMethodSafeTransferFrom,
+  ];
+
+  if (
+    currentConfirmation?.type &&
+    CONFIRMATIONS_WITH_NEW_HEADER.includes(currentConfirmation.type)
+  ) {
```

Comment:
Minor, would it make the logic clearer if we remove the nesting and have flat conditions for each alternate header?

We could even put the fallback into a local component such as `DefaultHeader`?